lib/compress/zstd_compress.c

```diff
@@ -2291,11 +2291,22 @@ static void ZSTD_copyBlockSequences(ZSTD_CCtx* zc)
             outSeqs[i].rep = 1;
             repIdx = i - outSeqs[i].offset;
 
+            /* Not first block */
             if (repIdx >= 0) {
-                outSeqs[i].offset = outSeqs[repIdx].offset;
-            }
-
-            if (repIdx == -1) {
+                /* Special case where litLength == 0 */
```

Comment:
This seems consistent with the docs to me. Please look at this carefully and let me know if I misunderstood.

Reply:
This isn't quite right, it will access out of bounds when `repIdx == 0`.

It might be simpler to do something like:

```c
if (outSeqs[i].litLength == 0) {
    if (outSeqs[i].offset < 3) {
        --repIdx;
    } else {
        repIdx = i - 1;
    }
}
assert(repIdx >= -3);
outSeqs[i].offset = repIdx >= 0 ? outSeqs[repIdx].offset : repStartValue[-repIdx - 1];
if (outSeqs[i].offset == 3) {
    --outSeqs[i].offset;
}
```